bridge/setu/broadcaster/broadcaster_test.go

```diff
@@ -1,61 +1,296 @@
 func TestBroadcastToHeimdall(t *testing.T) {
 	t.Parallel()
 
-	cdc := app.MakeCodec()
-	// cli context
-	tendermintNode := "tcp://localhost:26657"
-	viper.Set(helper.TendermintNodeFlag, tendermintNode)
+	viper.Set(helper.TendermintNodeFlag, dummyTenderMintNodeUrl)
 	viper.Set("log_level", "info")
-	// cliCtx := cliContext.NewCLIContext().WithCodec(cdc)
-	// cliCtx.BroadcastMode = client.BroadcastSync
-	// cliCtx.TrustNode = true
 
-	helper.InitHeimdallConfig(os.ExpandEnv("$HOME/.heimdalld"))
+	configuration := helper.GetDefaultHeimdallConfig()
+	configuration.TendermintRPCUrl = dummyTenderMintNodeUrl
+	configuration.HeimdallServerURL = dummyHeimdallServerUrl
+	helper.SetTestConfig(configuration)
+	helper.SetTestPrivPubKey(privKey)
+
+	mockCtrl := prepareMockData(t)
+	defer mockCtrl.Finish()
 
-	_txBroadcaster := NewTxBroadcaster(cdc)
+	testOpts := helper.NewTestOpts(nil, testChainId)
+	heimdallApp, sdkCtx, _ := createTestApp(false, testOpts)
+	testOpts.SetApplication(heimdallApp)
+	txBroadcaster := NewTxBroadcaster(heimdallApp.Codec())
+	txBroadcaster.CliCtx.Simulate = true
 
-	testData := []checkpointTypes.MsgCheckpoint{
-		{Proposer: hmTypes.BytesToHeimdallAddress(helper.GetAddress()), StartBlock: 0, EndBlock: 63, RootHash: hmTypes.HexToHeimdallHash("0x5bd83f679c8ce7c48d6fa52ce41532fcacfbbd99d5dab415585f397bf44a0b6e"), AccountRootHash: hmTypes.HexToHeimdallHash("0xd10b5c16c25efe0b0f5b3d75038834223934ae8c2ec2b63a62bbe42aa21e2d2d")},
-		{Proposer: hmTypes.BytesToHeimdallAddress(helper.GetAddress()), StartBlock: 64, EndBlock: 1024, RootHash: hmTypes.HexToHeimdallHash("0x5bd83f679c8ce7c48d6fa52ce41532fcacfbbd99d5dab415585f397bf44a0b6e"), AccountRootHash: hmTypes.HexToHeimdallHash("0xd10b5c16c25efe0b0f5b3d75038834223934ae8c2ec2b63a62bbe42aa21e2d2d")},
-		{Proposer: hmTypes.BytesToHeimdallAddress(helper.GetAddress()), StartBlock: 1025, EndBlock: 2048, RootHash: hmTypes.HexToHeimdallHash("0x5bd83f679c8ce7c48d6fa52ce41532fcacfbbd99d5dab415585f397bf44a0b6e"), AccountRootHash: hmTypes.HexToHeimdallHash("0xd10b5c16c25efe0b0f5b3d75038834223934ae8c2ec2b63a62bbe42aa21e2d2d")},
-		{Proposer: hmTypes.BytesToHeimdallAddress(helper.GetAddress()), StartBlock: 2049, EndBlock: 3124, RootHash: hmTypes.HexToHeimdallHash("0x5bd83f679c8ce7c48d6fa52ce41532fcacfbbd99d5dab415585f397bf44a0b6e"), AccountRootHash: hmTypes.HexToHeimdallHash("0xd10b5c16c25efe0b0f5b3d75038834223934ae8c2ec2b63a62bbe42aa21e2d2d")},
+	testCases := []struct {
+		name       string
+		msg        sdk.Msg
+		op         func(*app.HeimdallApp) error
+		expResCode uint32
+		expErr     bool
+		tearDown   func(*app.HeimdallApp) error
+	}{
+		{
+			name: "successful broadcast",
+			msg:  msgs[0],
+
+			op:         nil,
+			expResCode: 0,
+			expErr:     false,
+		},
+		{
+			name: "failed broadcast (insufficient funds for fees)",
+			msg:  msgs[1],
+			op: func(hApp *app.HeimdallApp) error {
+				acc := hApp.AccountKeeper.GetAccount(sdkCtx, hmTypes.BytesToHeimdallAddress([]byte(address)))
```

Comment:
minor one: the `[]byte` type conversion (used in many places in this file) is redundant

Reply:
Deduped: [89c97b2](https://github.com/maticnetwork/heimdall/pull/1170/commits/89c97b2b0a4f76edcdba870ffa0e9521614c90cf)